modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectRolesProcessor.java

```diff
@@ -0,0 +1,11 @@
+package com.enonic.xp.core.impl.project;
+
+import com.enonic.xp.project.Project;
+import com.enonic.xp.project.ProjectName;
+
+public interface ProjectRolesProcessor
+{
+    void processUpdate( final Project oldProject, final Project newProject );
```

Comment:
RolesProcessor could be replace with Command Pattern classes, so much spread in XP.
Not necessarily I like it, but it would be natural here, don't you agree?